mne/decoding/transformer.py

```diff
@@ -715,3 +715,168 @@ def transform(self, X):
+        See mne.filter.construct_iir_filter for details. If iir_params
+        is None and method="iir", 4th order Butterworth will be used.
+    verbose : bool, str, int, or None
+        If not None, override default verbose level (see mne.verbose).
+        Defaults to self.verbose.
+    """
+    def __init__(self, l_freq=None, h_freq=None, sfreq=1.0,
+                 filter_length='10s', l_trans_bandwidth=None,
+                 h_trans_bandwidth=None, n_jobs=1, method='fft',
+                 iir_params=None, verbose=None):
+        self.l_freq = l_freq
+        self.h_freq = h_freq
+        self.filter_length = filter_length
+        self.l_trans_bandwidth = l_trans_bandwidth
+        self.h_trans_bandwidth = h_trans_bandwidth
+        self.n_jobs = n_jobs
+        self.method = method
+        self.iir_params = iir_params
+        self.sfreq = sfreq
+
+        # check frequency params
+        float_params = dict([(self.l_freq, 'l_freq'), (self.h_freq, 'h_freq')])
+        for param in float_params.keys():
+            if not isinstance(param, (int, float)) and param is not None:
+                raise ValueError("%s must be of type int or float or None, "
+                                 "got %s instead" % (float_params[param],
+                                                     type(param)))
+
+        if not isinstance(self.sfreq, (int, float)):
+            raise ValueError("Sampling frequency must be int or float, "
+                             "got %s instead" % type(sfreq))
+
+        trans_bandwidth_params = dict([(self.l_trans_bandwidth,
+                                        'l_trans_bandwidth'),
+                                       (self.h_trans_bandwidth,
+                                        'h_trans_bandwidth')])
+        for param in trans_bandwidth_params.keys():
+            if not isinstance(param, (int, float)) and (param is not
+                                                        None) and (param !=
+                                                                   'auto'):
+                raise ValueError("%s must be of type int, float or None, "
+                                 "got %s instead" % (trans_bandwidth_params[
+                                     param], type(param)))
+
+        if not isinstance(self.filter_length, (int, str)) and (param is not
+                                                               None):
+            raise ValueError("filter_length must be a int, string or None. "
+                             "Got %s instead." % type(self.filter_length))
+
+        if not isinstance(self.n_jobs, int) and self.n_jobs == 'cuda':
+            raise ValueError('n_jobs must be int or "cuda", got %s instead.'
+                             % type(self.n_jobs))
+
+        if self.method not in ('fft', 'iir'):
+            raise ValueError('method must be either "fft" or "iir" or None. '
+                             'Got %s instead.' % self.method)
```

Comment:
You should now be able to use the `_triage_filter_params` to do all these (and more) checks for you.